webapp/channels/src/components/emoji/emoji_page.tsx

```diff
@@ -25,62 +25,52 @@ type Props = {
     };
 }
 
-export default class EmojiPage extends React.PureComponent<Props> {
-    static defaultProps = {
-        teamName: '',
-        teamDisplayName: '',
-        siteName: '',
-    };
-
-    componentDidMount() {
-        this.updateTitle();
-        this.props.actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
+export default function EmojiPage({teamDisplayName = '', teamName = '', siteName = '', scrollToTop, currentTheme, actions}: Props) {
+    useEffect(() => {
+        updateTitle();
+        actions.loadRolesIfNeeded(['system_admin', 'team_admin', 'system_user', 'team_user']);
         Utils.resetTheme();
-    }
 
-    componentWillUnmount() {
-        Utils.applyTheme(this.props.currentTheme);
-    }
+        return () => {
+            Utils.applyTheme(currentTheme);
+        };
+    }, [actions, currentTheme]);
 
-    updateTitle = () => {
-        document.title = Utils.localizeMessage('custom_emoji.header', 'Custom Emoji') + ' - ' + this.props.teamDisplayName + ' ' + this.props.siteName;
-    };
+    useEffect(() => {
+        updateTitle();
+    }, [siteName]);
 
-    componentDidUpdate(prevProps: Props) {
-        if (this.props.siteName !== prevProps.siteName) {
-            this.updateTitle();
-        }
-    }
+    const updateTitle = () => {
+        document.title = Utils.localizeMessage('custom_emoji.header', 'Custom Emoji') + ' - ' + teamDisplayName + ' ' + siteName;
+    };
 
-    render() {
-        return (
-            <div className='backstage-content emoji-list'>
-                <div className='backstage-header'>
-                    <h1>
-                        <FormattedMessage
-                            id='emoji_list.header'
-                            defaultMessage='Custom Emoji'
-                        />
-                    </h1>
-                    <AnyTeamPermissionGate permissions={[Permissions.CREATE_EMOJIS]}>
-                        <Link
-                            className='add-link'
-                            to={'/' + this.props.teamName + '/emoji/add'}
+    return (
+        <div className='backstage-content emoji-list'>
+            <div className='backstage-header'>
+                <h1>
+                    <FormattedMessage
+                        id='emoji_list.header'
+                        defaultMessage='Custom Emoji'
+                    />
+                </h1>
+                <AnyTeamPermissionGate permissions={[Permissions.CREATE_EMOJIS]}>
```

Comment:
Can we move this list to a constant outside the component so we are not passing a new list on every render?